frontend/ui/widget/textwidget.lua

```diff
@@ -46,14 +57,44 @@ function TextWidget:updateSize()
     end
     self._updated = true
 
+    -- Compute height:
+    -- Used to be:
+    --   self._height = math.ceil(self.face.size * 1.5)
+    --   self._baseline_h = self._height*0.7
+    -- But better compute baseline alignment from freetype font metrics
+    -- to get better vertical centering of text in box
+    -- (Freetype doc on this at https://www.freetype.org/freetype2/docs/tutorial/step2.html)
+    local face_height, face_ascender = self.face.ftface:getHeightAndAscender()
+    self._height = math.ceil(face_height) + 2*self.padding
+    self._baseline_h = Math.round(face_ascender) + self.padding
+    -- With our UI fonts, this usually gives 0.72 to 0.74, so text is aligned
+    -- a bit lower than before with the hardcoded 0.7
+
     -- In case we draw truncated text, keep original self.text
     -- so caller can fetch it again
     self._text_to_draw = self.text
 
+    if self.text and type(self.text) ~= "string" then
+        self.text = tostring(self.text)
+    end
+    self._is_empty = false
+    if not self.text or #self.text == 0 then
+        self._is_empty = true
+        self._length = 0
+        return
+    end
+
+    -- Compute width:
+    if self.use_xtext then
+        self:_measureWithXText()
+        return
+    end
+
+    -- Only when not self.use_xtext:
+
     -- Note: we use kerning=true in all RenderText calls
     --- @todo Don't use kerning for monospaced fonts. (houqp)
```

Comment:
This still relevant?

Reply:
Not with the use_xtext code (Haven't checked, but I hope HB does the right thing with monospace fonts.
But for the legacy code, it might (it would still use kerning with monospaced fonts).

Reply:
OK, I'll remove the `@todo` and keep that comment in parenthesis. As we'll probably don't have to fix it if we keep using xtext.